### PR TITLE
Add bomb option.

### PR DESCRIPTION
**Pull request by anthony-geoghegan**
Windows PowerShell requires a BOM for scripts encoded with UTF-8.

I came across this problem when a function that I had declared in my user profile (`$PROFILE`) started printing `â€˜` instead of `‘` and `â€™` instead of  `’`.  See also <https://stackoverflow.com/q/14482253/1640661>

I had originally tested for the (file) encoding like so:

    if &encoding == 'utf-8'
      setlocal bomb
    endif

However, after thinking about it, I concluded this wasn’t necessary since setting the `bomb` option won’t have any effect – unless `fileencoding` is set to

> "utf-8", "ucs-2", "ucs-4" or one of the little/big endian variants.

---

By the way, I don’t know what the purpose of `ftplugin/ps1xml.vim` is but I updated that file as well as `ftplugin/ps1.vim `

### CODE REVIEW

File: ftplugin/ps1xml.vim [hunk 1]
```diff
@@ -14,6 +14,11 @@ let b:did_ftplugin = 1
 setlocal tw=0
 setlocal commentstring=#%s
 setlocal formatoptions=tcqro
+" Enable autocompletion of hyphenated PowerShell commands,
+" e.g. Get-Content or Get-ADUser
+setlocal iskeyword+=-
```
Comment (heaths) [inline ftplugin/ps1xml.vim]: This is already handled elsewhere in the script and seems unrelated to the request in this PR.

Comment (anthony-geoghegan) [inline ftplugin/ps1xml.vim]: Hi heaths. I wasn't sure about that because I'd never worked with `.ps1xml` files (and I presumed that they should have the same configuration). Now that I've researched what `.ps1xml` files are and how they're used, I realise I was wrong to copy those lines from `.ps1`.

Would it be easier if I resubmit the pull request – or should I just leave it as it is?

Comment (heaths) [inline ftplugin/ps1xml.vim]: Just push a new commit. It's just a pain if you abandon this and open a new one. All information about this is lost. So fix it, commit, and push all in the same branch from which you created this PR in the first place.

Comment (anthony-geoghegan) [inline ftplugin/ps1xml.vim]: Hi Heath,

This is the first time, I’ve submitted a PR that had issues with it so I’m still figuring out the GitHub review workflow. I’ve fixed the issue and pushed a new commit so I hope it’s OK now.

Thanks for your patience, understanding (and diligence),
Anthony

**Comment by heaths** [comment]
Tested this out a bit and makes sense. Thanks!

**Comment by anthony-geoghegan** [comment]
Thanks. It's great to be able to contribute positively and it was a good learning experience to see how faults in the pull request are dealt with (and can be rectified). I'm understanding more how GitHub has become so popular as a platform for open source software. Its UI provides collaboration features that are much easier to use than sending a patch to a mailing list.

**Comment by cwegener** [comment]
This change does not seem right at all.

Windows is using UTF16 for most operations, so you should simply set your `.editorconfig` for your project to use utf16-be or utf16-le for .ps1 files and this will fix your specific problem.

Forcing every user of vim-ps1 to use a non-recommended file encoding (i.e. UTF8 with BOM) does not seem like the right solution to your problem at all!

Also, regarding the two specific Unicode codepoints that you provided (`U+2018` and `U+2019`), from my PowerShell experience, these are very rarely used. And they also have 'non-curly' counterparts which work in ASCII.

May I ask what purpose you are using the curly quotes for? Are you generating some sort of documentation that uses curly quotes?
